### PR TITLE
refactor!: remove faction camps

### PR DESCRIPTION
**Pull request by Dread-Pirate-Hogarth**
## Purpose of change

https://github.com/cataclysmbnteam/Cataclysm-BN/discussions/3866

## Describe the solution
delete

didn't try to preserve savegames. discussion said we didn't need to, so I didn't. don't think you could find anyone that'd do that stupid amount of work.

should build but haven't done so yet

### CODE REVIEW

**Comment by RoyalFox2140** [comment]
This must've been a monumental amount of work to do. I hope everything goes off without a hitch.

**Comment by Dread-Pirate-Hogarth** [comment]
> This must've been a monumental amount of work to do. I hope everything goes off without a hitch.

was much faster to do from the web interface, did a check afterwards to see if any errors were reported. must've missed some stuff

**Comment by Lamandus** [comment]
Please try to fix the errors soonish

File: src/condition.cpp [hunk 1]
```diff
@@ -393,18 +392,6 @@ void conditional_t<T>::set_at_om_location( const JsonObject &jo, const std::stri
...
-            }
-            // legacy check
-            const std::string &omt_str = omt_ref.id().c_str();
-            return omt_str.find( "faction_base_camp" ) != std::string::npos;
-        } else if( location == "FACTION_CAMP_START" ) {
-            return !recipe_group::get_recipes_by_id( "all_faction_base_types",
-                    omt_ref.id().c_str() ).empty();
-        } else {
             return omt_ref == oter_id( oter_no_dir( oter_id( location ) ) );
         }
```
Comment (scarf005) [inline src/condition.cpp]: i think closing bracket should be removed too.

**Review by scarf005 — CHANGES_REQUESTED** [review]
hi, thanks for the hard work!

however, there's many issues preventing the PR from building, namely:

- many files like map.cpp and mapgen.cpp still references basecamp (e.g `#include "basecamp.h"` or `bcp`)
- (faction camp) bulletin board function missing in iexamine.cpp
- faction.cpp has unwrapped brackets which breaks function signature in `npc::faction_display`
- also unwrapped brackets in `condition.cpp`
- `job_data` unremoved
- missing `companion_mission`. i think it's maybe used for other cases than faction camp?

could you make sure the PR can be built locally and mark it as ready?

**Comment by Dread-Pirate-Hogarth** [comment]
no need to run tests yet i have it in vs

will tell you when it compiles

**Comment by Dread-Pirate-Hogarth** [comment]
image


didn't see any use for companion_mission outside of camp but will double check

**Comment by Dread-Pirate-Hogarth** [comment]
companion_mission only gets used at the board for camp mission
not much use buying land for your camp that doesn't exist

regular talk_mission_list still works

npc go_to you from radio is broken but seems that was broken before the PR

**Comment by scarf005** [comment]
one major save-breaking issue: npcs sent scavenging from faction camp before the faction camp removal will be lost. @Dread-Pirate-Hogarth maybe should we take care of this.

**Comment by Dread-Pirate-Hogarth** [comment]
they should load now. will be placed wherever they were when sent on mission

will still have junk data for npc_companion_mission but this is not a big issue. data will not impact performance and can be removed later or kept for use in new systems

**Comment by scarf005** [comment]
image

they do return correctly now.

**Review by scarf005 — APPROVED** [review]
it errors with invalid zone type on load but can be easily resolved by removing said zone. overall LGTM but would like to wait for @Coolthulhu's opinion.

**Comment by Lamandus** [comment]
> it errors with invalid zone type on load but can be easily resolved by removing said zone. overall LGTM but would like to wait for @Coolthulhu's opinion.

But only one error, or all the time?

**Comment by scarf005** [comment]
it'll keep error on load until all existing camp related zones are removed.

image

image

**Comment by Lamandus** [comment]
Then loot-zones.json has to be changed, too

**Review by Lamandus — CHANGES_REQUESTED** [review]
Changes to the loot zones have to be made. Not only in the json, I suppose in the source code, too.

**Comment by Lamandus** [comment]
> it'll keep error on load until all existing camp related zones are removed.
> 
> image
> 
> image
> 
> strangely, only `CAMP_FOOD` will error; `basecamp: storage` won't.

I suppose it is because storage is "just food storage". While camp food is linked to a function.

**Comment by Dread-Pirate-Hogarth** [comment]
ill take a look

**Comment by Dread-Pirate-Hogarth** [comment]
simple preexisting error in implementation of iterator. if zone exists in loaded world, should error once and properly erase invalid zone now.

**Comment by Lamandus** [comment]
> simple preexisting error in implementation of iterator. if zone exists in loaded world, should error once and properly erase invalid zone now.

gonna take a look now, I am compiling it, will take a few minutes for testing.

**Comment by Dread-Pirate-Hogarth** [comment]
i pushed another fix right after you posted. make sure you check that one

**Review by Lamandus — APPROVED** [review]

**Comment by scarf005** [comment]
image

this causes segmentation fault on loading old save.

**Pull request merged**